Added Build Instructions for Linux with Raspberry Pi

RChestnut opened the pull request:
Add new instruction on /docs/ for Debian-Based Linux running on Raspberry Pi and other devices with ARM64

RChestnut committed:
Added Build Instructions for Linux with Raspberry Pi

RChestnut committed:
Fixed some mis-typo

RChestnut committed:
Minor changes on title.

RChestnut committed:
Fixed mis-typo and grammars.

RChestnut committed:
fix mis-typo

RChestnut committed:
Change description about UFW

RChestnut committed:
Change Title

RChestnut commented:
Last mis-typo has been fixed. Dunno that i can spend more hours for checking these and fixing.

NamecoinGithub commented on docs/build-linux-rpi.md:
```diff
@@ -0,0 +1,440 @@
+For starting, You should go to the LLL-TAO folder that you compiled the daemon. and input <code>nexus</code> and enter gently.
+
+Or if you registered the Nexus daemon as environment variable, just input <code>nexus</code> anywhere.
+
```
./nexus=START   CTRL-C=STOP

not everyone will install nexus into usr/bin

In that case ./ is required to start nexus node ./nexus

RChestnut replied:
This method works same as <code>./nexus</code> for fresh installing because there's no environment variable added before and moved to the build directory already. Anyway, For the standard, I'm gonna fix it. Thanks :D

RChestnut replied:
This had applied on commit <code>11a46f3</code>.

NamecoinGithub commented:
Might as well add the Nexus Quantum Encryption Algo Falcon into the Readme, as far as I know it's NOT documented anywhere at the moment!
nexus.conf

falcon=1

RChestnut commented:
@NamecoinGithub I'm gonna fix it today with other mistypos that i found after i arrive at home, with adding API function for terminating the daemon safely.

By the way, The falcon encryption is needed hard forking to version 6.0, as @VidereLicet said on issue #69. So It will be added after hard fork update.
(Seems that <code>falcon=1</code> is for testnet. Maybe after this encryption algorithm have been updated with hard forking, This could be default like what tritium base did that had been on hard fork before.)

RChestnut committed:
Fixed and Update the instruction

- Fixed mistypos.
- Changed <code>nexus</code> to <./nexus> on starting up daemon.
- Changed default RPC port as 9334 from 9888, According to Nexus Resource Hub.
- Adding instructions of API function.

RChestnut commented:
Fixed mistypos and applied some updates on commit <code>11a46f3</code>, Including changing the method of calling daemon with <code>nexus</code> to <code>./nexus</code>.

Adding <code>falcon=1</code> on the instruction of writting <code>nexus.conf</code> that suggested by @NamecoinGithub is dismissed.
(Please check the reply about it and issue #69)

RChestnut committed:
Update current version to 5.1.5-rc14-9

RChestnut committed:
Fixed mistypo

line 6: and -> on

RChestnut committed:
Fixed mistypo

NamecoinGithub reviewed (changes requested):
https://github.com/Nexusoft/LLL-TAO/blob/merging/docs/build-linux.md

You had that part Correct with ONLY 
nexus

./nexus is how you start the Node 
that is Part 5 

I didn't want to have you change anything except ADD in the additional 
./nexus
way to start the NODE
Calling
nexus 
Will start the node also, but only if Global installed!
I just wanted you to document Both

falcon=1 WORKS
That Request is for something different
Nexus Quantum Encryption WORKS for the Signature Key Pad
falcon=1
Is how you Activate it to use!  Otherwise it's the Default ECC keys or something 1-Time Keypad!

NamecoinGithub commented:
Nexus uses Falcon512 Default

RChestnut commented:
> https://github.com/Nexusoft/LLL-TAO/blob/merging/docs/build-linux.md
>

I dunno what is the purpose about you quoted this document.

If you want to say to make me modify this document, That document wasn't originated from me.

Or if you quoted this for reference, I didn't use this but i did with 'Nexus Resource Hub's one.

> You had that part Correct with ONLY nexus
> 
> ./nexus is how you start the Node that is Part 5
> 
> I didn't want to have you change anything except ADD in the additional ./nexus way to start the NODE Calling nexus Will start the node also, but only if Global installed! I just wanted you to document Both


I had written that thing on commit <code>11a46f3</code> already, with adding as;
> Or if you registered the Nexus daemon as environment variable, Just input <code>nexus</code> anywhere.

on the Part 5.

Also had written on Part 2, About how to add the daemon to environment variable, like this;
> Optional: You may add the <code>nexus</code>daemon as environment variable by coping it to <code>/usr/bin</code>, For calling daemon on globally.
> ...
> If you don't, Old Nexus daemon will be called from <code>/usr/bin</code> and the version that you newly built will be ignored to be called.

You can check these by <code>git diff</code> with inputting comparible versions.

> falcon=1 WORKS That Request is for something different Nexus Quantum Encryption WORKS for the Signature Key Pad falcon=1 Is how you Activate it to use! Otherwise it's the Default ECC keys or something 1-Time Keypad!

Also, About the <code>falcon=1</code>, I had sent a DM to @VidereLicet with some questions about it.

So I will handle this after I receive the message from him.

RChestnut commented:
#### `UPDATE : Seems that i need to pause updating this instruction.`
-----------------------------------------
I found that <code>ARM32</code>(but only <code>armv7l</code>) is supported after PR #52.

So I have needed to update the instruction and have some research, But seems that something weird on architecture recognizing logic.

On <code>makefile.cli</code>:
>
```
else ifeq (${ARM64}, 1) # For ARM64(aarch64), It needs the 'ARM64=1' flag for recognizing architecture.
        BUILD_ARCH=aarch64
        CXXFLAGS+= -march=native -mtune=native

else ifeq ($(shell uname -m),armv7l) # But for ARM32(armv7l), It crawls from 'uname -m' function.
        BUILD_ARCH=arm
        # Also there's no CXXFLAGS for tweaking the compiling task.
```
>

As the code on this <code>makefile.cli</code> currently, Seems that the recognizing logic between <code>ARM32</code> and <code>ARM64</code> are different, as <code>ARM64</code> requires the flag(<code>'ARM64=1'</code>) but <code>ARM32</code> doesn't.

It may feels that the method of <code>ARM32</code> is much convenience, But it has a serious issue; **This logic could recognize only the <code>'armv7l'</code> architecture as 32-bit.**

Yes, it means that this logic cannot be recognizes other 32-bit environment such as <code>'armv8l'</code>, Which is the 32-bit linux distribution running on the <code>ARM64(aarch64)</code> architecture. (Currently using on Raspberry Pi 3B and above.)

Also there's no backward support, such as <code>'armv6l'</code>, Other 32-bit ARM that used on RPi1.
(Ofc, I know that this is very old device that out-dated. But shouldn't we at least give users a warning that this architecture is not supported?)

Anyway, I need to wait the message from Colin about some question with <code>'falcon=1'</code> that i had asked via DM, So there's plenty time to open the issue and make patch for this.

Until this issues are completed, I will pause updating this instruction, gonna open the issue about it and make some hotfixes.

RChestnut commented:
<img src="https://github.com/user-attachments/assets/b038e045-7ea7-4f8d-bdf6-8ffb79c9273e" width="30%" height="30%"  />

`(You can press the picture to see in full size)`

> falcon=1 WORKS
That Request is for something different
Nexus Quantum Encryption WORKS for the Signature Key Pad
falcon=1
Is how you Activate it to use! Otherwise it's the Default ECC keys or something 1-Time Keypad!
>

> Nexus uses Falcon512 Default



I received a reply about <code>'falcon=1'</code> flag from @VidereLicet today. And here's what he said about this;
> @VidereLicet: **<code>falcon=1</code>** enables **`Falcon-1024`**, default uses `brainpool` keys but sigchain arch protects even that from QC ***so falcon isn't default until a bit down the road.***
I have a signature pruning technique that will only require two signatures to maintain sigchain security no matter how long so ***when this is implemented we will enable falcon by default and drop brainpool***.
>

It seems same context that i said to you before on previous comment.
> By the way, The falcon encryption is needed hard forking to version 6.0, as @VidereLicet said on issue #69. So It will be added after hard fork update.
(Seems that <code>falcon=1</code> is for testnet. Maybe after this encryption algorithm have been updated with hard forking, This could be default like what tritium base did that had been on hard fork before.)
>

EDIT : I had asked about does this flag being used on mainnet may cause the orphan blocks. 
And i've got a message from him on early morning of today, he confirmed that it won't make the orphans on being used in mainnet.

But it wasn't not necessary for running nodes.

> @VidereLicet: Falcon is ready, works fine. When I implement pruning it cleans up all old too so can use it now if you'd like to.
>
>No falcon will not cause any orphan blocks.

So since this document is for the beginners, Might the treatment of this flag at here is not suitable.

Also may the definition of this flag is needed to be treated in the other new document properly, but not here. (such as `'nexus.conf Flags Documentation'`)

Because it is not for giving the description about the all of flags that used in <code>'nexus.conf'</code> 

And it will be as default on the next hard fork, so may it consider that the flag will not be used also.

@NamecoinGithub So, Adding <code>'falcon=1'</code> on the instruction as you suggested will be dismissed.(again)

----------------------------------------------------
Anyways, The issue related with 32-bit ARM build was   opened - #208

Although opening the issue, This instruction is ready for merging - only for 64-bit now.

If this instruction is merged before i try to fix the issue, I'm gonna update the instruction with adding 32-bit compatibility after fix that issue and open the PR again!

RChestnut committed:
Merge branch 'Nexusoft:merging' into merging-guide-rpi
